### PR TITLE
feat: support for different encoders

### PR DESCRIPTION
**Pull request by mattstrom**
specifically for AWS API Gateway both for running in the browser and in Node

### CODE REVIEW

File: src/encoders.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  }
+}
+
+/**
+ * @param {'arraybuffer' | 'aws'} socketType
+ * @returns {EncoderSet}
+ */
+export function getEncoderSet (socketType) {
+  const isBrowser = typeof window !== 'undefined'
+
```
Comment (idaunis) [inline src/encoders.js:106]: I think it would be simpler to check `socketType === 'arraybuffer'` first since is common

File: src/encoders.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    /*
+     * The result looks like
+     * "data:application/octet-stream;base64,<your base64 data>",
+     * so we split off the beginning:
+     */
+    return base64Url.substring(base64Url.indexOf(',') + 1)
+  },
+  /** @param {any} data */
+  responseDecoder: (data) => {
+    let arr = new Uint8Array(data)
```
Comment (idaunis) [inline src/encoders.js]: maybe this can also be simplified so we don't double assign

File: src/encoders.js (hunk 1 above)
Comment (mattstrom) [reply src/encoders.js:106]: Good call. Previously I had `defaultEncoderForBrowser` and `defaultEncoderForNode` before realizing they were identical

File: src/encoders.js (hunk 2 above)
Comment (mattstrom) [reply src/encoders.js]: Example? I pulled this from your original code, but it does feel awkward

**Review by idaunis — APPROVED** [review]

**Pull request merged**